### PR TITLE
✨ Adding Support for Mutating DNSPolicy

### PR DESCRIPTION
**Pull request by christopherhein**
**What this PR does / why we need it**:
This adds a feature gate that allows us to mutate the pods dnsPolicy using a label, removing the mutation that disallows setting `ClusterFirst` DNS.

Signed-off-by: Chris Hein <me@chrishein.com>

### CODE REVIEW

File: virtualcluster/pkg/syncer/conversion/mutate.go [hunk 1]
```diff
@@ -172,7 +173,10 @@ func PodMutateDefault(vPod *v1.Pod, saSecretMap map[string]string, services []*v
 		if err != nil {
 			return err
 		}
-		mutateDNSConfig(p, vPod, vc.Spec.ClusterDomain, nameServer)
+		allowDNSPolicyMutation := featuregate.DefaultFeatureGate.Enabled(featuregate.TenantAllowDNSPolicyMutation)
+		if _, set := vPod.GetLabels()[constants.TenantDisableDNSPolicyMutation]; !allowDNSPolicyMutation && !set {
```
Comment (Fei-Guo) [inline virtualcluster/pkg/syncer/conversion/mutate.go:177]: The naming looks confusion. We call `mutateDNSConfig` when `allowDNSPolicyMutation` is false?

Comment (christopherhein) [reply virtualcluster/pkg/syncer/conversion/mutate.go:177]: good point, let me change that.

File: virtualcluster/pkg/syncer/conversion/mutate.go [hunk 2]
```diff
@@ -172,7 +173,10 @@ func PodMutateDefault(vPod *v1.Pod, saSecretMap map[string]string, services []*v
 		if err != nil {
 			return err
 		}
-		mutateDNSConfig(p, vPod, vc.Spec.ClusterDomain, nameServer)
+		allowTenantDNSPolicy := featuregate.DefaultFeatureGate.Enabled(featuregate.TenantAllowDNSPolicy)
+		if _, set := vPod.GetLabels()[constants.TenantDisableDNSPolicyMutation]; !allowTenantDNSPolicy && !set {
```
Comment (Fei-Guo) [inline virtualcluster/pkg/syncer/conversion/mutate.go:177]: I think it should be `!(allowTenantDNSPolicy && set)` ?

Comment (christopherhein) [reply virtualcluster/pkg/syncer/conversion/mutate.go:177]: Isn't this logic the same? `&&` covers that, unless you mean stylistically.

Comment (Fei-Guo) [reply virtualcluster/pkg/syncer/conversion/mutate.go:177]: The mutate call is skipped if someone sets the constants.TenantDisableDNSPolicyMutation label even if the feature gate is disabled.

Comment (christopherhein) [reply virtualcluster/pkg/syncer/conversion/mutate.go:177]: Oh duh 🤦

**Review by weiling61 — APPROVED** [review]
/lgtm

**Pull request merged**